x-pack/plugins/canvas/canvas_plugin_src/functions/common/compare.ts

```diff
@@ -49,22 +49,26 @@ export const compare: ContextFunctionFactory<'compare', Context, Arguments, bool
         return a !== b;
       case 'lt':
         if (typesMatch) {
-          return !!a && !!b && a < b;
+          // @ts-ignore #35433 This is a wonky comparison for nulls
```

Comment:
```js
const typesMatch = a !== null && typeof a === typeof b;
```

Wouldn't doing that above fix this check? If you get a null context, then the comparisons where it would matter all fail (which I think is correct).

Reply:
What about if b is null?

The problem up until now is that using `null` in a gt/lt scenario works, it's simply forbidden by `eslint` in a TS scenario, (and rightly so).  So converting the logic here so that it runs as it has in the past was getting really heavy, (hence opening a task for a follow-up).